src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsRepository.cs

```diff
@@ -151,17 +160,23 @@ public override async Task<string[]> GetSecretSnapshots(ScriptSecretsType type,
             // Prefix is secret blob path without extension
             string prefix = Path.GetFileNameWithoutExtension(GetSecretsBlobPath(type, functionName)) + $".{ScriptConstants.Snapshot}";
 
-            BlobResultSegment segmentResult;
+            var blobList = new List<BlobItem>();
             try
             {
-                segmentResult = await Container.ListBlobsSegmentedAsync(string.Format("{0}/{1}", _secretsBlobPath, prefix.ToLowerInvariant()), null);
+                var segmentResult = Container.GetBlobsAsync(prefix: string.Format("{0}/{1}", _secretsBlobPath, prefix.ToLowerInvariant()));
+                var asyncEnumerator = segmentResult.GetAsyncEnumerator();
+
+                while (await asyncEnumerator.MoveNextAsync())
```

Comment:
await foreach?

Reply:
Ideally, we would use await foreach here but it's not supported in the C# version that functions host targets (async streams added in C# 8.0). 
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md#flat-listing

Reply:
You can set the lang version in the project file (https://www.meziantou.net/using-foreach-with-ienumerator-t-iasyncenumerator-t-in-csharp-9.htm) that's what we do for our libraries even though we target .net standard